lg_activity/src/lg_activity/activity.py

```diff
@@ -260,6 +268,52 @@ def _is_activity_active(self):
             self.callback(self.topic, state=False, strategy='activity')
             return False
 
+    def _is_message_active(self):
+        """keep last message. Empty list returns False TODO add other cases, empty string, null, ...""" 
+        if self.messages:
+             try:
+                 self.messages = [self.messages[-1]]  # skip to last message and keep it
+                 if self.messages[0]['overlays']:  # TODO add set False methods, else True
+                     self.callback(self.topic, state=True, strategy='message', delay=1)
+                     logger.debug("Setting strems state True")
+                     return True
+             except (IndexError, KeyError, ValueError, TypeError) as e:
+                 logger.debug("error checking stream messages: %" % e)
+        self.messages = []
+        self.callback(self.topic, state=False, strategy='message')
+        logger.debug("Streams are off")
+        return False
+
+    def _is_duration_active(self):
+        """check scene message for duration or own state if no message"""
+        if not self.tracker.active:  # ignore attract_loop playing
+            self.messages = []
+            return False
+        if self.messages:
+            try:
+                duration = int(self.messages[-1]['message.duration'])  # skip to last message duration
```

Comment:
is the key called `message.duration`? or is it `messages[-1]['message']['duration']` (which again use `.get('message', {}).get('duration', 0)` just in case.

Reply:
the acitvity "slot" creates the key "message.duration", I dind't see any other way to use the slot, we could skip it and just import the json message
I will add the default values either way